csharp/src/Google.Protobuf/UnsafeCollectionOperations.cs

```diff
@@ -0,0 +1,88 @@
+#region Copyright notice and license
+
+// Protocol Buffers - Google's data interchange format
+// Copyright 2008 Google Inc.  All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#endregion
+
+using System;
+using System.Security;
+using Google.Protobuf.Collections;
+
+namespace Google.Protobuf;
+
+/// <summary>
+/// An unsafe class that provides a set of methods to access the underlying data representations of
+/// collections.
+/// </summary>
+[SecuritySafeCritical]
+public static class UnsafeCollectionOperations
+{
+    /// <summary>
+    /// <para>
+    /// Returns a <see cref="Span{T}"/> that wraps the current backing array of the given
+    /// <see cref="RepeatedField{T}"/>.
+    /// </para>
+    /// <para>
+    /// Values in the <see cref="Span{T}"/> should not be set to null. Use
+    /// <see cref="RepeatedField{T}.Remove(T)"/> or <see cref="RepeatedField{T}.RemoveAt(int)"/> to
+    /// remove items instead.
+    /// </para>
+    /// <para>
+    /// The returned <see cref="Span{T}"/> is only valid until the <see cref="RepeatedField{T}"/> is
+    /// modified, after which its state becomes undefined.
+    /// </para>
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of elements in the <see cref="RepeatedField{T}"/>.
+    /// </typeparam>
+    /// <param name="field">
+    /// The <see cref="RepeatedField{T}"/> for which to wrap the current backing array.
```

Comment:
Specify "Must not be null."

Reply:
(We haven't historically been terribly good at this in the proto library, but I'd like to try to make sure that at least new code is clearer...)